test/scripts/run-api-tests.js

```diff
@@ -54,6 +54,7 @@ const runAllTests = async (args) => {
       FORCE_COLOR: 1,
       ENV_PATH: process.env.ENV_PATH,
       JWT_SECRET: 'aSecret',
+      STRAPI_GRAPHQL_V4_COMPATIBILITY_MODE: 'true',
```

Comment:
Good catch, it either means we should do one of these things:

- add "graphql" to the list of restricted config filenames / make it an official file
- make it a feature flag, so use something like STRAPI_FEATURES_GRAPHQL_V4COMPATIBILITYMODE (and update the FeaturesConfig type to support something other than 'future' so it can be placed in features.js at `graphql: { v4CompatibilityMode }`
-  fit it into an existing config file, like STRAPI_SERVER_GRAPHQL_V4COMPATIBILITYMODE so that we could pick it up automatically with config.get('server.graphql.v4CompatibilityMode')

Note that other than renaming it, we don't have to do the actual work in this PR, it can be added later as long as we don't have to break it by changing its name. So it's basically a choice of which way we plan to go, and then calling it something based on that.

My personal preference is make it a feature flag but any of those work for me.

Reply:
No need to do all that I added that flag only for the tests for now until I refactor them and I'll remove it afterwards.